src/main/scala/firrtl/ir/IR.scala

```diff
@@ -22,24 +24,73 @@ case object NoInfo extends Info {
   override def toString: String = ""
   def ++(that: Info): Info = that
 }
-case class FileInfo(info: StringLit) extends Info {
-  override def toString: String = " @[" + info.serialize + "]"
+
+/** Stores the string of a file info annotation in its escaped form. */
+case class FileInfo(escaped: String) extends Info {
+  override def toString: String = " @[" + escaped + "]"
   //scalastyle:off method.name
   def ++(that: Info): Info = if (that == NoInfo) this else MultiInfo(Seq(this, that))
+  def unescaped: String = FileInfo.unescape(escaped)
+}
+
+object FileInfo {
+  @deprecated("Use FileInfo.fromUnEscaped instead. FileInfo.apply will be removed in FIRRTL 1.5.", "FIRRTL 1.4")
+  def apply(info: StringLit): FileInfo = new FileInfo(escape(info.string))
+  @deprecated("FileInfo.unapply with StringLit will be removed in FIRRTL 1.5.", "FIRRTL 1.4")
+  def unapply(arg: FileInfo): Option[StringLit] = if(arg.eq(null)) None else Some(StringLit(arg.unescaped))
+  def fromEscaped(s: String): FileInfo = new FileInfo(s)
+  def fromUnescaped(s: String): FileInfo = new FileInfo(escape(s))
+  /** prepends a `\` to: `\`, `\n`, `\t` and `]` */
+  def escape(s: String): String = EscapeFirrtl.translate(s)
+  /** removes the `\` in front of `\`, `\n`, `\t` and `]` */
+  def unescape(s: String): String = UnescapeFirrtl.translate(s)
+  /** take an already escaped String and do the additional escaping needed for Verilog comment */
+  def escapedToVerilog(s: String) = EscapedToVerilog.translate(s)
+
+  // custom `CharSequenceTranslator` for FIRRTL Info String escaping
+  type CharMap = (CharSequence, CharSequence)
+  private val EscapeFirrtl = new LookupTranslator(Seq(
+    "\\" -> "\\\\" : CharMap,
+    "\n" -> "\\n",
+    "\t" -> "\\t",
+    "]" -> "\\]"
+  ).toMap.asJava)
+  private val UnescapeFirrtl = new LookupTranslator(Seq(
+    "\\\\" -> "\\" : CharMap,
+    "\\n" -> "\n",
+    "\\t" -> "\t",
+    "\\]" -> "]"
+  ).toMap.asJava)
+  // EscapeFirrtl + EscapedToVerilog essentially does the same thing as running StringEscapeUtils.unescapeJava
+  private val EscapedToVerilog = new AggregateTranslator(
+    new LookupTranslator(Seq(
+      // ] is the one character that firrtl needs to be escaped that does not need to be escaped in
+      "\\]" -> "]" : CharMap,
+      "\"" -> "\\\"",
+      // \n and \t are already escaped
+      "\b" -> "\\b",
+      "\f" -> "\\f",
+      "\r" -> "\\r",
+  ).toMap.asJava),
```

Comment:
@jackkoenig : any idea why this wouldn't compile with Scala `2.11` but work fine with `2.12`?
```
[error] /Volumes/develop/firrtl/src/main/scala/firrtl/ir/IR.scala:74:3: illegal start of simple expression
[error]   ).toMap.asJava),
[error]   ^
[error] /Volumes/develop/firrtl/src/main/scala/firrtl/ir/IR.scala:78:1: ')' expected but '}' found.
[error] }
[error] ^
[error] two errors found
```

Reply:
Trailing comma on the previous line?

Reply:
I think I may have found what it was upset about: the trailing `,`